_studio/mfx_lib/vpp/include/mfx_vpp_hw.h

```diff
@@ -908,6 +908,8 @@ namespace MfxHwVideoProcessing
             mfxFrameSurface1 *pInputSurface,
             mfxFrameSurface1 *pOutputSurface);
 
+        mfxStatus IsCopyPassThroughPreferred(bool &preferred) const;
```

Comment:
This design looks ugly, this function should simply return ```bool```

Reply:
I don't like this function:). Both prototype (+1 to @mgonchar) and how it named/what it does. It returns not deterministic result: "I check copy pass thru, maybe I like it maybe not - decide later". I would like to see function named useSOMETHING() with the obvious inclination.

To be honest, I got lost in CopyPassThrough name. What it actually mean? Copy with memcpy or CM?

Reply:
I tried to say something like "performance of CopyPassThrough is good, you want to use it" vs "it's better to omit CopyPassThrough" and to hide low-level details what is used inside and what is faster.